vault/policy_test.go

```diff
@@ -494,3 +494,63 @@ path "foo/+*" {
 		t.Errorf("bad error: %s", err)
 	}
 }
+
+func TestPolicy_BadParameterWildcards(t *testing.T) {
+	type testCase struct {
+		name          string
+		rules         string
+		expectedError string
+	}
+
+	testCases := []testCase{{
+		name: "partial wildcard",
+		rules: `
+			path "/" {
+				capabilities = ["read"]
+				allowed_parameters = {
+					"zip*" = []
+				}
+			}`,
+		expectedError: `path "/": allowed_parameters: parameter key "zip*" cannot contain partial wildcards`,
+	}, {
+		name: "wildcard in a nested parameter",
+		rules: `
+			path "/" {
+				capabilities = ["read"]
+				denied_parameters = {
+					"/a/*/b" = []
+				}
+			}`,
+		expectedError: `path "/": denied_parameters: nested parameter key "/a/*/b" cannot contain non-tail wildcards`,
```

Comment:
Just curious on this. Was this too difficult to support or just didn't make sense in this context?

Reply:
This is not something natively supported by `pointerstructure` syntax and I could not see too many usecases where it would be useful. I suppose, in theory it is possible to expand the syntax further in the future to support non-tail wildcards, but the implementation might not be very straight forward.

Reply:
got it! makes sense